hack/lib/dev-env.sh

```diff
@@ -0,0 +1,182 @@
+# Admin kubeconfig generated by kind
+kind_kubeconfig="${client_dir}/kind-kubeconfig.yaml"
+
+function create_network() {
+    if [[ -n $(docker network ls | grep authenticator-dev) ]]; then
+        echo "${NETWORK_NAME} exists"
+    else
+        echo "Creating network ${NETWORK_NAME}"
+        docker network create "${NETWORK_NAME}" \
+            --driver bridge \
+            --subnet "${NETWORK_SUBNET}" \
+            --opt "com.docker.network.bridge.enable_ip_masquerade=true"
+    fi
+}
+
+function delete_network() {
+    if [[ -n $(docker network ls | grep authenticator-dev) ]]; then
+        echo "Deleting network ${NETWORK_NAME}."
+        docker network rm "${NETWORK_NAME}"
+    else
+        echo "Network ${NETWORK_NAME} doesn't exist, doing nothing."
+    fi
+}
+
+function write_authenticator_config() {
+    mkdir -p "${authenticator_config_host_dir}"
+    sed -e "s|{{ADMIN_ARN}}|${ADMIN_ARN}|g" \
+        -e "s|{{AUTHENTICATOR_STATE_DIR}}|${authenticator_state_dest_dir}|g" \
+        -e "s|{{AUTHENTICATOR_GENERATE_KUBECONFIG}}|${authenticator_generate_kubeconfig}|g" \
+        -e "s|{{AUTHENTICATOR_KUBECONFIG}}|${authenticator_kubeconfig}|g" \
+        -e "s|{{AUTHENTICATOR_PORT}}|${AUTHENTICATOR_PORT}|g" \
+        -e "s|{{AUTHENTICATOR_IP}}|${AUTHENTICATOR_IP}|g" \
+        -e "s|{{CLUSTER_NAME}}|${CLUSTER_NAME}|g" \
+        "${authenticator_config_template}" > "${authenticator_config}"
+}
+
+function write_kind_config() {
+    mkdir -p "${kind_config_host_dir}"
+    sed -e "s|{{AUTHENTICATOR_EXPORT_HOST_DIR}}|${authenticator_export_host_dir}|g" \
+        -e "s|{{APISERVER_CONFIG_DEST_DIR}}|${apiserver_config_dest_dir}|g" \
+        -e "s|{{APISERVER_AUTH_WEBHOOK_KUBECONFIG}}|${apiserver_auth_webhook_kubeconfig}|g" \
+        -e "s|{{CLUSTER_NAME}}|${CLUSTER_NAME}|g" \
+        "${kind_config_template}" > "${kind_config}"
+}
+
+function start_authenticator() {
+    mkdir -p "${authenticator_state_host_dir}"
+    mkdir -p "${authenticator_export_host_dir}"
+    chmod 777 "${authenticator_state_host_dir}"
+    chmod 777 "${authenticator_export_host_dir}"
+    docker run \
+        --detach \
+        --ip "${AUTHENTICATOR_IP}" \
+        --mount "type=bind,src=${authenticator_config_host_dir},dst=${authenticator_config_dest_dir}" \
+        --mount "type=bind,src=${authenticator_state_host_dir},dst=${authenticator_state_dest_dir}" \
+        --mount "type=bind,src=${authenticator_export_host_dir},dst=${authenticator_export_dest_dir}" \
+        --name aws-iam-authenticator \
+        --network "${NETWORK_NAME}" \
+        --publish ${authenticator_healthz_port}:${authenticator_healthz_port} \
+        --publish ${AUTHENTICATOR_PORT}:${AUTHENTICATOR_PORT} \
+        --rm \
+        "${AUTHENTICATOR_IMAGE}" \
+        server \
+        --config "${authenticator_config_dest_dir}/authenticator.yaml"
+}
+
+function kill_authenticator() {
+    docker kill aws-iam-authenticator || true
+}
```

Comment:
the issue I have with this is that it's not actually testing the installation/operating practice for `aws-iam-authenticator` that is described in the README: deploy the authenticator as a Daemonset on each control plane node in the k8s cluster, using [the example manifest](https://github.com/kubernetes-sigs/aws-iam-authenticator/blob/master/deploy/example.yaml). Why not just kubectl apply a slightly-modified version of the example YAML files against the KinD cluster? That would do dual service of validating the recommended installation approach for the authenticator.

Reply:
Yeah.  The reason I went with the above approach is because I decided to let the authenticator generate its own certificate and the webhook kubeconfig on startup.  Since the path to this config needs to be passed to the apiserver as a flag on startup, this suggests starting the authenticator before the kind cluster, because the API server is started automatically on cluster creation and is configured in the kind cluster configuration.  That being said, we can just use the [init](https://github.com/kubernetes-sigs/aws-iam-authenticator/blob/master/cmd/aws-iam-authenticator/init.go) command to pre-generate the configuration, and then launch authenticator, and then patch the API server.

Reply:
We should try the approach that allows quicker iteration that does not need a tear down and restart of the kind cluster. In both approaches i think we get it, probably bit easier with daemonset approach. 
The second criteria for choosing is closeness to setups. I'm not sure which one is more prevalent, but i'd lean towards the approach implemented due to my familiarity with managed setups.